added a frequency2scale function as the dual to scale2frequency

tkuraku opened the pull request:
Hi, I added a frequency2scale function as the dual of scale2frequency. In my own use of PyWavelets I find this function useful and felt it would be a nice addition.

grlee77 commented:
Thanks @tkuraku, it seems reasonable to provide this function. Can you document it in
https://github.com/PyWavelets/pywt/blob/master/doc/source/ref/cwt.rst
as well?

tkuraku commented:
@grlee77 Thanks for considering my pull request. I have added some documentation and squashed my commits. take a look and let me know if you have any requests.

grlee77 reviewed (approved):
Thanks, @tkuraku. This looks good. I just made a small edit to explain the normalization in the function docstrings themselves as well.
